Readme.md

```diff
@@ -45,6 +45,15 @@ sent to the client (optional)
 * `contentType`: If the content to be sent is anything other than
 `application/json`, then the `contentType` property must be set (optional)
 * `bearerType`: string specifying the Bearer string (optional)
+* `function auth (key) {}` : this function will test if a key is valid. If the key is valid it
+must return a truthy value  or a promise that resolves to a truthy value. If function resolves to
+falsey values (false,null,undefined,0) authentication will fail with 401 value. If function fails
+or promise rejects authentication will fail (see `failSilent`). If `auth` is a function, `keys` will 
+be ignored. If `auth` is not a function or undefined, `keys` will be used.  
+* `failSilent`: if set to true and `auth` is specified any error or rejection occurred while calling `auth`
+will be returned with a 401 status and a invalid authorization header. If set to false it will return a 
+500 status with the error or the rejection. 
```

Comment:
This is not if the authentication is failed (wrong token). This is for example if you do a query to the db and the db is down. In this case the proper response is a 500 (server error) but some people would like to return a 401 in any case.

Reply:
Then I would mandate that the function _always_ return a literal `true` or `false`, where the `false` would result in a `401`. If the function throws an error, return a `500`. This is the simplest implementation.

Another implementation would require the function to return an `{error, code, value}` response. If the key checks out then it would be `{value: true}`. If the key is invalid, `{value: false}`. And if some error occurs, `{error: Error('oops'), code: 500}`. This would allow the implementor to designate which code they really want to return.

Reply:
I like both options. That is: 
* return true or false (literals) or return {error, code, value}
* any throws or rejects => error 500 
* no failsSilent option 
* any value not listed in 1 => error 500

If that's ok I can do it.

Reply:
I think it would be easier to implement, document, and use to pick only one of the styles. Either is okay with me.

Reply:
Ok I implement, document and made tests for style of literal booleans.